packages/core/data-transfer/src/strapi/remote/handlers.ts

```diff
@@ -76,21 +82,55 @@ export const createTransferHandler =
         };
 
         const teardown = (): server.Payload<server.EndMessage> => {
+          clearTimeout(updateUserAbilityTimer);
           delete state.controller;
           delete state.transfer;
-
           return { ok: true };
         };
 
-        const init = (msg: client.InitCommand): server.Payload<server.InitMessage> => {
+        // TODO: set a maximum time allowed to pass without a successful userAbility ability
+
+        /**
+         * update userAbility
+         */
+        const updateUserAbility = async () => {
+          // TODO: clean this up using getService if possible OR create typings for PermissionEngineService instead of using `any`
+          userAbility = await strapi
+            .service<any>('admin::permission')
+            .engine.generateUserAbility(user);
+          // only restart timer if connection is still open, in case something went wrong in teardown
+          if (openStates.includes(ws.readyState)) {
+            updateUserAbilityTimer = setTimeout(updateUserAbility, updateAbilityMilliseconds);
+          }
+        };
+        /**
+         * start a timer that updates userAbility as long as the connection is alive
+         */
+        const initializeAbilityTimer = () => {
```

Comment:
@Convly Would it actually be possible to use a Strapi hook to listen for role/user updates?